pkg/orderedmap/convert.go

```diff
@@ -70,20 +70,21 @@ func (c Conversion) fromUnorderedMaps(object interface{}) interface{} {
 		panic("Expected map[interface{}]interface{} instead of *unordered.Map in fromUnorderedMaps")
 
 	case []interface{}:
+		result := make([]interface{}, len(typedObj))
 		for i, item := range typedObj {
-			typedObj[i] = c.fromUnorderedMaps(item)
+			result[i] = c.fromUnorderedMaps(item)
 		}
-		return typedObj
+		return result
 
 	// some sources (e.g. toml library) yield specific type of slices.
 	// slices in Go are not covariant, so each flavor of slice must have its own case, here.
 	// process these exactly the same way we do for generic slices (prior case)
 	case []map[string]interface{}:
-		resultArray := make([]interface{}, len(typedObj))
+		result := make([]interface{}, len(typedObj))
```

Comment:
Renamed this because all the other case blocks use `result` as the new variable name